samples/bluetooth/central_hr/README.rst

```diff
@@ -9,6 +9,9 @@ Overview
 Similar to the :ref:`Central <bluetooth_central>` sample, except that this
 application specifically looks for heart-rate monitors and reports the
 heart-rate readings once connected.
+If TRACK_VND=1 is passed to cmake build system, the application looks for
```

Comment:
Not sure we need this, why not have it always scan for the vendor UUID? OR better yet create a dedicated application that works as conterpart to peripheral sample like we have central_hr.

Reply:
There are two applications implementing the Central role. Central and Central_HR. The first one scans everything, finds any peer and if it is close enough just connects to it and does nothing else. It does not deal with attributes, with GATT, nor encryption.
The application Central_HR does a complex GATT characteristic scanning, searching for devices that advertise their support. If it finds one, it subscribes to its notifications. The patch here does something similar: finds a vendor specific service (which is already defined by the peripheral sample which export both HRS and a vendor specific characteristic) and then, instead of subscribing, reads it for a number of times.
Why we need this? There is no application that uses encrypted connections to create more complex traffic scenarios. Peripheral_sc_only does only pairing. All other standard GATT services are unencrypted. Only the vendor specific attribute exposed in the peripheral sample is encrypted but is not read by any central application.

Reply:
Central_HR is the central side implementation of the Heart Rate Service. It is often used as the "hello world" of bluetooth services due to it's simplicity. Central_hr and peripheral_hr should also match up.

As long as the VND handling is in the peripheral sample then I think the right place for the central side of VND would be the central sample.

Passing configuration options directly to the cmake build system is not something we do. We usually use Kconfig for this, but samples don't have kconfig. They should only have one configuration. Either we should remove the TRACK_VND config parameter, or create two samples.

Using VND service to show encryption is fine. The peripheral_hids has HoG (HID over GATT) which requires authentication. But that service might be a bit more complicated so might as well use the VND.